theano/tensor/nnet/sigm.py

```diff
@@ -352,7 +352,7 @@ def c_code(self, node, name, inp, out, sub):
 
         # float16 limits: -17.0, 6.0
         # We use the float32 limits for float16 for now as the
-        # computation will happend in float32 anyway.
+        # computation will happen in float32 anyway.
         if (node.inputs[0].type == scalar.float32 or
                 node.inputs[0].type == scalar.float16):
             return """%(z)s = %(x)s < -103.0f ? 0.0 : %(x)s > 14.0f ? %(x)s : log1p(exp(%(x)s));""" % locals()
```

Comment:
The GPU also uses this C code.  But it shouldn't be required after my changes in this PR and the next to convert the internal graph of Composite to float32 in GpuElemwise.